observability/observability.go

```diff
@@ -0,0 +1,92 @@
+package observability
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+
+	"github.com/TBD54566975/ftl/internal/log"
+	ftlv1 "github.com/TBD54566975/ftl/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/protos/xyz/block/ftl/v1/ftlv1connect"
+	"github.com/alecthomas/atomic"
+	"github.com/alecthomas/errors"
+
+	"github.com/alecthomas/types"
+
+	"github.com/bufbuild/connect-go"
+)
+
+type Observability struct {
+	registrationFailure atomic.Value[types.Option[error]]
```

Comment:
@alecthomas I think we need this here for the `Ping` method below, but it's not being initialized anywhere. Might need to update this before we merge but everything seems to be working normally.

Reply:
The default (zero) value is valid, so it doesn't need to be initialised if you don't want to.